Gioco Dama

Lorenzo1997 opened the pull request:
Multiplayer 2v2
Single player contro il bot
Documentazione gioco

cesare-montresor committed:
Create release.yml

FrancescoMasaia committed:
[ADD]

workflow for release binaries

FrancescoMasaia committed:
Update release.yml

FrancescoMasaia committed:
Update release.yml

FrancescoMasaia committed:
Update release.yml

FrancescoMasaia committed:
Update release.yml

FrancescoMasaia committed:
Update release.yml

FrancescoMasaia committed:
Update release.yml

FrancescoMasaia committed:
Update release.yml

FrancescoMasaia committed:
Update release.yml

FrancescoMasaia committed:
Update release.yml

FrancescoMasaia committed:
Update release.yml

FrancescoMasaia committed:
Update release.yml

FrancescoMasaia committed:
Update release.yml

Lorenzo1997 committed:
Gioco Dama

Multiplayer 2v2
Single player contro il bot
Documentazione gioco

dariost commented on src/games/dama/instance.rs:
```diff
@@ -0,0 +1,119 @@
+        // Creo il vettore che conterrà il percorso valido
+        let mut percorso_valido: Vec<String> = Vec::new();
+
+        // Inizzia sempre la partita il secondo giocatore che si conette
```
Inizzia -> Inizia

dariost commented on src/games/dama/instance.rs:
```diff
@@ -0,0 +1,119 @@
+                _ = spectators.write(("Turno nero!\n").as_bytes()).await;
+
+                // Controllo chi deve muovere le pedine nere 
+                if &giocatore[1].name == "ServerBot$0"
```
Non puoi assumere che il bot si chiami sempre in quel modo

dariost reviewed (changes requested):
Faccio un po' di commenti generali perché ci sono cose che si ripetono più volte:

- la rappresentazione interna dello stato è molto strana (usi un char al posto di un enum), e porta a scrivere codice che è molto poco idiomatico;
- l'interazione è più quella che ti aspetteresti con un umano che con un programma, in particolare esula dagli standard posti dagli altri giochi già implementati;
- una cosa fondamentale che non deve mai fare il manager (e speranzosamente anche il bot) è crashare, ma nel codice ci sono svariati punti dove il programma può andare in panic (vedesi tutti gli unwrap).

Come requisito minimo per mergiare questa PR in un branch di staging ti chiedo di risolvere i commenti puntuali che ho fatto e di risolvere almeno il terzo punto generale (quindi levare tutti gli unwrap, gestendo gli errori nei casi in cui possono capitare, e sostituendolo con expect nei casi in cui matematicamente non può crashare).

Lorenzo1997 committed:
Cambio unwrap con expect

Lorenzo1997 committed:
Verifica bot ottimizzata

Ora se il nome del bot cambia il gioco capirà comunque che sta muovendo il bot e non un player

Lorenzo1997 commented:
Correzioni apportate:
1- Sostituiti tutti gli unwrap con expect
2- Corretto errore di ortografia
3- Ottimizzato il controllo di chi deve muovere le pedine tra bot e player

dariost commented on src/games/dama/logic.rs:
```diff
@@ -0,0 +1,795 @@
+        // Pulisco la variabili altrimenti si portano dietro tutti i valori precedenti
+        percorso.clear();
+        mosse.clear();
+        giocatore.input.read_line(&mut percorso).await.expect("Lettura della riga fallita !!!");
```
Questa lettura può fallire, e va gestita

dariost commented on src/games/dama/instance.rs:
```diff
@@ -0,0 +1,119 @@
+                _ = spectators.write(("Turno bianco!\n").as_bytes()).await;
+
+                // Controllo chi deve muovere le pedine bianche 
+                if giocatore[0].name.starts_with("ServerBot$")
```
Non puoi assumere niente sul bot, per te deve essere come un qualunque altro giocatore. Chi è il bianco e chi è il nero deve essere o determinato in modo random e comunicato ai giocatori, oppure scelto tramite un qualche parametro a priori.

Lorenzo1997 committed:
Merge branch 'master' into dama

Lorenzo1997 committed:
Verifica abbandono della partita

Lorenzo1997 committed:
Comunicazione dati al bot tramite input e output

Lorenzo1997 commented:
Sono stati corretti tutti i punti come richiesto.

Lorenzo1997 committed:
Trattamento del bot come un normale giocatore terminato

Lorenzo1997 commented:
Correzione del bot terminata.

Lorenzo1997 committed:
Utilizzo di una sola notazione, aggiunta parametri pace e timeout

Lorenzo1997 committed:
Aggiornamento gel progetto all'ultima versione

Lorenzo1997 committed:
Correzione errori di battitura

Lorenzo1997 commented:
Aggiornamento del repo e correzioni codice fatte

dariost commented:
This PR has been squash-merged in commit f60a9503a97e5bbab8c60e27abbcdccb854ec845
